sqlglot/expressions.py

```diff
@@ -5037,7 +5037,13 @@ def to_table(sql_path: t.Optional[str | Table], **kwargs) -> t.Optional[Table]:
     if not isinstance(sql_path, str):
         raise ValueError(f"Invalid type provided for a table: {type(sql_path)}")
 
-    catalog, db, table_name = (to_identifier(x) for x in split_num_words(sql_path, ".", 3))
+    catalog, db, table_name, *rest = (
+        t.cast(t.Optional[Expression], to_identifier(x)) for x in split_num_words(sql_path, ".", 3)
+    )
+
+    if rest and table_name:
+        table_name = Dot.build(t.cast(t.List[Expression], [table_name, *rest]))
```

Comment:
This produces the same AST as the one we'd get if we tried to `parse_one` a table with >3 parts:

```python
>>> import sqlglot
>>> sqlglot.parse_one("select * from a.b.c.d.e")
(SELECT expressions:
  (STAR ), from:
  (FROM this:
    (TABLE this:
      (DOT this:
        (DOT this:
          (IDENTIFIER this: c, quoted: False), expression:
          (IDENTIFIER this: d, quoted: False)), expression:
        (IDENTIFIER this: e, quoted: False)), db:
      (IDENTIFIER this: b, quoted: False), catalog:
      (IDENTIFIER this: a, quoted: False))))
>>> sqlglot.exp.to_table("a.b.c.d.e")
(TABLE this:
  (DOT this:
    (DOT this:
      (IDENTIFIER this: c, quoted: False), expression:
      (IDENTIFIER this: d, quoted: False)), expression:
    (IDENTIFIER this: e, quoted: False)), db:
  (IDENTIFIER this: b, quoted: False), catalog:
  (IDENTIFIER this: a, quoted: False))
```